frontend/src/Order.tsx

```diff
@@ -18,44 +18,45 @@ import { FieldHeader } from './components/FieldHeader'
 const Order = () => {
     const api = new apiBackend()
 
+    const [
+        { exClass, country, accessories, userType, wbs, deliveryAddress, nIpads, deviceType, simType, userShortnames, orderResponsible },
+        setFormState,
+    ] = useState<OrderForm>(initialFormState)
+
     const [isSubmitEnabled, setIsSubmitEnabled] = useState(false)
     const [resultRitm, setResultRitm] = useState('')
     const [isSubmitPopupOpen, setIsSubmitPopupOpen] = useState(false)
-    const [selectedExClass, setSelectedExClass] = useState('')
-    const [selectedCountry, setSelectedCountry] = useState('Norway')
+
     const [countryList, setCountryList] = useState<string[]>([])
-    const [selectedAccessories, setSelectedAccessories] = useState<string[] | undefined>(['Charger Plug', 'Neck Strap'])
-    const [selectedUserType, setSelectedUserType] = useState('Equinor personnel')
-    const [wbs, setWbs] = useState('')
-    const [address, setAddress] = useState('')
-    const [ipadCount, setIpadCount] = useState('')
+
     const [numberOfiPadsError, setNumberOfiPadsError] = useState(false)
-    const [deviceType, setDeviceType] = useState('personal')
-    const [radioCheckedSIM, setRadioCheckedSIM] = useState('wifi')
-    const [shortname, setShortname] = useState('')
+
     const [validPositions, setValidPositions] = useState<PositionDetails[]>([])
-    const [selectedPositionId, setSelectedPositionId] = useState('')
-    const [hasFetched, setHasFetched] = useState(false)
+    const [hasFetchedPositions, setHasFetchedPositions] = useState(false)
 
-    const exClassOptions = createDropdownOptions(exClasses, selectedExClass)
-    const userTypeOptions = createDropdownOptions(userTypes, selectedUserType)
+    const exClassOptions = createDropdownOptions(exClasses, exClass)
+    const userTypeOptions = createDropdownOptions(userTypes, userType)
 
-    const mandatoryFields = [selectedCountry, selectedPositionId, wbs, address, selectedExClass, selectedUserType, ipadCount]
+    const mandatoryFields = [country, orderResponsible, wbs, deliveryAddress, exClass, userType, nIpads]
+
+    const setSingleField = (name: string, value: any) => {
+        setFormState(prevState => ({ ...prevState, [name]: value }))
+    }
 
     useEffect(() => {
         api.getCountries().then(response => {
             setCountryList(response.sort())
         })
     }, [])
-    const countryDropdown = createDropdownOptions(countryList, selectedCountry)
+    const countryDropdown = createDropdownOptions(countryList, country!)
```

Comment:
Why is there an exclamation mark here?

Reply:
Because country on OrderForm can be null